### PR TITLE
Deprecate Session helper.

### PR DESCRIPTION
**Pull request by dereuromark**
Rebased version of https://github.com/cakephp/cakephp/pull/5448

### CODE REVIEW

**Comment by ADmad** [comment]
Don't remove the test file. It should be removed when the class itself is actually removed.

**Comment by ADmad** [comment]
Hmm but since the constructor trigger's error keeping test file would mean builds would fail, so it has to be removed :(

Other tests are failing due to use of SessionHelper which need to be updated.

**Comment by dereuromark** [comment]
That's why I deleted it. Should be all good now.

File: src/View/Helper/SessionHelper.php [hunk 1]
```diff
@@ -41,6 +43,18 @@ class SessionHelper extends Helper
...
 
     /**
+     *  Constructor
+     *
+     * @param \Cake\View\View $View The View this helper is being attached to.
+     * @param array $config Configuration settings for the helper.
+     */
+    public function __construct(View $View, array $config = [])
+    {
+        trigger_error('SessionHelper has been deprecated. Use request->session() instead.', E_USER_WARNING);
```
Comment (markstory) [inline src/View/Helper/SessionHelper.php]: Should be E_USER_DEPRECATED. E_WARNING is often on in production applications.

**Comment by dereuromark** [comment]
I also squashed the commits.

**Comment by dereuromark** [comment]
Bump

**Comment by dereuromark** [comment]
Oh, we merged this into 3.0 (master) and not 3.1. Should we fix that?

**Comment by dereuromark** [comment]
And if anyone is too lazy to replace the helper calls, he could easily use the [Shim.Session helper](https://github.com/dereuromark/cakephp-shim/blob/3.0/src/View/Helper/SessionHelper.php) as drop in replacement:

```php
// in AppController
public $helpers = ['Shim.Session'];
```

Works out of the box.

**Comment by ADmad** [comment]
@dereuromark We maintain BC and the trigger_error() message won't be shown with debug off so there's no problem adding deprecations in bugfix release.

**Comment by dereuromark** [comment]
OK, I corrected the version: https://github.com/cakephp/cakephp/commit/f36cfc351e3f8bffb4877206773da9360a59682c
And made a docs correction for 3.0

**Pull request merged**